sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-jedis/src/main/java/com/azure/messaging/eventhubs/checkpointstore/jedis/JedisRedisCheckpointStore.java

```diff
@@ -0,0 +1,151 @@
+        String prefix = prefixBuilder(fullyQualifiedNamespace, eventHubName, consumerGroup);
+        try (Jedis jedis = jedisPool.getResource()) {
+            Set<String> members = jedis.smembers(prefix);
+            ArrayList<PartitionOwnership> list = new ArrayList<>();
+            for (String member : members) {
+                //get the associated JSON representation for each for the members
+                String partitionOwnershipJSON = jedis.hmget(member, "partitionOwnership").get(0);
+                //convert JSON representation into PartitionOwnership
+                try {
+                    PartitionOwnership partitionOwnership = jacksonAdapter.deserialize(partitionOwnershipJSON, PartitionOwnership.class, SerializerEncoding.JSON);
+                    list.add(partitionOwnership);
+                }
+                catch (IOException e) {
+                    throw LOGGER.logExceptionAsError(Exceptions
+                        .propagate(e)); }
+            }
+            jedisPool.returnResource(jedis);
+            return Flux.fromStream(list.stream());
+        }
+    }
+
+    /**
+     * This method updates the checkpoint in the Jedis resource for a given partition.
+     *
+     * @param checkpoint Checkpoint information for this partition
+     * @return Null
+     */
+    @Override
+    public Mono<Void> updateCheckpoint(Checkpoint checkpoint) {
+        if (!isCheckpointValid(checkpoint)) {
+            throw LOGGER.logExceptionAsWarning(Exceptions
+                .propagate(new IllegalStateException(
+                    "Checkpoint is either null, or both the offset and the sequence number are null.")));
+        }
+        String prefix = prefixBuilder(checkpoint.getFullyQualifiedNamespace(), checkpoint.getEventHubName(), checkpoint.getConsumerGroup());
+        String key = keyBuilder(prefix, checkpoint.getPartitionId());
+        try (Jedis jedis = jedisPool.getResource()) {
+            //Case 1: new checkpoint
+            if (!jedis.exists(prefix) || !jedis.exists(key)) {
+                jedis.sadd(prefix, key);
+                try {
+                    jedis.hset(key, "checkpoint", jacksonAdapter.serialize(checkpoint, SerializerEncoding.JSON));
+                } catch (IOException e) {
+                    throw LOGGER.logExceptionAsError(Exceptions
+                        .propagate(e));
+                }
+            }
+            //TO DO: Case 2: checkpoint already exists in Redis cache
+            jedisPool.returnResource(jedis);
+        }
+        return null;
+    }
+    private String prefixBuilder(String fullyQualifiedNamespace, String eventHubName, String consumerGroup) {
+        return fullyQualifiedNamespace + "/" + eventHubName + "/" + consumerGroup;
+    }
+    private String keyBuilder(String prefix, String partitionId) {
+        return prefix + "/" + partitionId;
+    }
+    private Boolean isCheckpointValid(Checkpoint checkpoint) {
```

Comment:
nit: add empty new lines between your method signatures, it makes it easier to read.